lib/redis/commands/streams.rb

```diff
@@ -65,14 +65,31 @@ def xadd(key, entry, approximate: nil, maxlen: nil, nomkstream: nil, id: '*')
       #   redis.xtrim('mystream', 1000)
       # @example With options
       #   redis.xtrim('mystream', 1000, approximate: true)
-      #
-      # @param key         [String]  the stream key
-      # @param mexlen      [Integer] max length of entries
-      # @param approximate [Boolean] whether to add `~` modifier of maxlen or not
+      # @example With strategy
+      #   redis.xtrim('mystream', '1-0', strategy: 'MINID')
+      #
+      # @overload xtrim(key, maxlen, strategy: 'MAXLEN', approximate: true)
+      #   @param key         [String]  the stream key
+      #   @param maxlen      [Integer] max length of entries
+      #   @param strategy    [String]  the limit strategy, must be MAXLEN
+      #   @param approximate [Boolean] whether to add `~` modifier of maxlen or not
+      #   @param limit       [Integer] maximum count of entries to be evicted
+      # @overload xtrim(key, minid, strategy: 'MINID', approximate: true)
+      #   @param key         [String]  the stream key
+      #   @param minid       [String]  minimum id of entries
+      #   @param strategy    [String]  the limit strategy, must be MINID
+      #   @param approximate [Boolean] whether to add `~` modifier of minid or not
+      #   @param limit       [Integer] maximum count of entries to be evicted
       #
       # @return [Integer] the number of entries actually deleted
-      def xtrim(key, maxlen, approximate: false)
-        args = [:xtrim, key, 'MAXLEN', (approximate ? '~' : nil), maxlen].compact
+      def xtrim(key, len_or_id, strategy: 'MAXLEN', approximate: false, limit: nil)
+        strategy = strategy.to_s.upcase
+        unless %w[MAXLEN MINID].include?(strategy)
+          raise ArgumentError, "strategy must be MINID or MAXLEN"
+        end
```

Comment:
I'm not convinced all this validation is worth it, might as well directly pass it to the server.

Reply:
Apparently, we are going to have to either compact/handle the nils in this test case or change the behavior:
https://github.com/redis/redis-rb/blob/master/test/lint/streams.rb#L131 . If we don't then, the we receive a TypeError with this message `Unsupported command argument type: NilClass`.

A couple of options:
- Change the behavior so that we throw the TypeError instead of a Redis::CommandError
- Leave as-is, and compact one
- Combine your suggestion above with a compact before we call `send_command`

What's your preference?

Reply:
Updated in 5d7ceea00524d287affb272f8afc791c803b66bc